lib/pbench/server/database/models/users.py

```diff
@@ -101,16 +102,25 @@ def evaluate_email(self, key: str, value: str) -> str:
         valid = validate_email(value)
         return valid.email
 
+    def add_token(self, token: ActiveTokens):
+        """Add the given token to the database, removing any expired tokens
+
+        Args:
+            token : An ActiveTokens object add for this user
+        """
+        try:
+            self.auth_tokens.append(token)
+            Database.db_session.add(token)
+            Database.db_session.commit()
+        except Exception:
+            Database.db_session.rollback()
+            raise
+
     def update(self, **kwargs):
         """Update the current user object with given keyword arguments."""
         try:
             for key, value in kwargs.items():
-                if key == "auth_tokens":
-                    # Insert the auth token
-                    self.auth_tokens.append(value)
-                    Database.db_session.add(value)
-                else:
```

Comment:
We don't want to keep some code here which ensures that we don't try to "update" the `auth_tokens` thingie?

Reply:
We'll just get an exception.